frontend/web/src/app/components/physical-print/physical-print-list/physical-print-list.component.ts

```diff
@@ -0,0 +1,163 @@
+import {AfterViewInit, Component, EventEmitter, Input, OnInit, Output, ViewChild} from '@angular/core';
+import {Router} from '@angular/router';
+import {BsModalRef, BsModalService} from 'ngx-bootstrap';
+import {PhysicalPrint} from '../../../model/physical-print';
+import {PhysicalPrintService} from '../../../services/physical-print/physical-print.service';
+import {ErrorService} from '../../../services/error.service';
+declare var $: any;
+
+@Component({
+  selector: 'app-physical-print-list',
+  templateUrl: './physical-print-list.component.html',
+  styleUrls: ['./physical-print-list.component.scss'],
+  providers: [PhysicalPrintService, ErrorService]
+})
+export class PhysicalPrintListComponent implements OnInit {
+  private modalRef: BsModalRef;
+  @ViewChild('modalDelete') modalDelete;
+  private toBeDeleted: number = null;
+
+  private table;
+  @Input('physicalPrint') physicalPrints: PhysicalPrint [];
+  @Output() selected: EventEmitter<PhysicalPrint> = new EventEmitter<PhysicalPrint>();
+
+
+  constructor(private physicalPrintService: PhysicalPrintService,
+              private errorService: ErrorService,
+              private router: Router,
+              private modalService: BsModalService) {
+  }
+
+  ngOnInit() {
+    this.table = $('#table');
+  }
+
+  ngAfterViewInit(): void {
+    this.loadAndPopulate();
+  }
+
+  loadAndPopulate() {
+    /* get users */
+    this.physicalPrintService.getAllPhysicalPrint().subscribe(
+      (physicalPrints) => {
+        this.physicalPrints = physicalPrints;
+
+        this.populate();
+        this.prepareTriggers();
+      }, (error) => {
+        alert(error.verbose_message);
+      },
+    );
+  }
+
+  private populate() {
+    const _self = this;
+    /*manually generate columns*/
+    const columns = [ {
+      title: 'ID',
+      field: 'id',
+      sortable: true,
+    },{
+      title: 'Digital Print',
+      field: 'digitalPrintId',
+      sortable: true,
+    },{
+      title: 'SLM File',
+      field: 'slmPath',
+      sortable: true,
+    }, {
+      field: 'operate',
+      title: '<span class="glyphicon glyphicon-cog">',
+      align: 'center',
+      events: {
+        'click .edit': function(e, value, row, index) {
+          _self.router.navigate([_self.router.url, row.id]);
+        },
+        'click .delete': function(e, value, row, index) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Expected method shorthand in object literal ('{'click .delete'() {...}}'). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS3498)